Add an initial implementation of the DST-parser (#1)
soramitsu/dst-parser: README.md

> # Example docstring parser

Reviewer: ```suggestion
# Docstring Parser
```

Author: I am not as sure about this part, the parser is intended specifically for our example code and it may look more vague, but I'll add that and rename it if I have a better name later.

Reviewer: you can say which code base and documentation it was designed for, it can be "docstring parser for Iroha tutorial"

Reviewer: anyway my main point is that it's not an "example" :) we can sure come up with a better name for this but just doctring parser should also be ok, and adding a paragraph on what this was created for will help clear up any confusion

> A source code docstring parser for a custom C-like (including C++, JavaScript, Rust)
> and Python-like syntax, where documentation sections are marked in comments using
> `FRAGMENT` and `ESCAPE` statements.

Reviewer: ```suggestion
The docstring parser for C-like (including C++, JavaScript, Rust)
and Python-like syntax, where code that should be included in documentation is marked with custom comments.
```

## Syntax

> A text is considered a named fragment when it is located between `// BEGIN FRAGMENT: <name>` and `// END FRAGMENT`, `<>` not included. This syntax is case-sensitive.

Reviewer: ```suggestion
A piece of code is considered a named fragment when it is located between `// BEGIN FRAGMENT: <name>` and `// END FRAGMENT`, `<>` not included. This syntax is case-sensitive.
```

Don't use special characters in names; `snakeCase` is preferrable.

Fragments can be included in one another. In that case the lines matching `// BEGIN FRAGMENT: <name>` and `// END FRAGMENT` are removed.

Elements between `// BEGIN ESCAPE` and `// END ESCAPE` are excluded unconditionally from the tutorial.

## Installation

* `npm install dst-parser --save-dev` if you want to install it as a development dependency of your package.

> * `npm install dst-parser -g` to install the parser globally.

Reviewer: * Install the parser as a development dependency of your package:

  ```
  npm install dst-parser --save-dev
  ```

* Install the parser globally:

  ```
  npm install dst-parser -g
  ```

Author: Thanks, added

# Docstring Parser

The docstring parser for C-like (including C++, JavaScript, Rust)
and Python-like syntax, where code that should be included in documentation is marked with custom comments.

## Syntax

A piece of code is considered a named fragment when it is located between `// BEGIN FRAGMENT: <name>` and `// END FRAGMENT`, `<>` not included. This syntax is case-sensitive.

Don't use special characters in names; `snakeCase` is preferrable.

Fragments can be included in one another. In that case the lines matching `// BEGIN FRAGMENT: <name>` and `// END FRAGMENT` are removed.

Elements between `// BEGIN ESCAPE` and `// END ESCAPE` are excluded unconditionally from the tutorial.

## Installation

Install the parser as a development dependency of your package:

```bash
npm install dst-parser --save-dev
```

Install the parser globally:

```bash
npm install dst-parser -g
```

## Example code

> ```rust

Reviewer 2: ```suggestion
```ts
```

Reviewer: Isn't the example in Rust though?
